### ISSUE TITLE
SpringBootCondition in OAuth2SsoDefaultConfiguration and OAuth2SsoCustomConfiguration

### ISSUE THREAD

**Issue by jarnbjo** [spring-projects/spring-boot#6783]
Found with Spring Boot 1.3.6, but also applies to the current snapshot in git.

If I understand it correctly, the `EnableOAuth2Sso` annotation imports two conditional configurations, of which one is applied:
- `OAuth2SsoCustomConfiguration` should be used if the `@EnableOAuth2Sso` annotation is used on a subclass of `WebSecurityConfigurerAdapter`. The logic for this condition is implemented in `OAuth2SsoCustomConfiguration.WebSecurityEnhancerCondition`.
- `OAuth2SsoDefaultConfiguration` is used otherwise, if the `EnableOAuth2Sso` annotation is used elsewhere and _not_ on a subclass of `WebSecurityConfigurerAdapter`. The logic for this condition is implemented in `OAuth2SsoDefaultConfiguration.NeedsWebSecurityCondition`.

Both condition implementations are very similar, the only difference is that they return the opposite outcome of eachother: match/noMatch. They ask the bean factory for all beans having an EnableOAuth2Sso annotation, iterate through the beans and check if any of them extends `WebSecurityConfigurerAdapter`.

If we now have a configuration class similar to this:

``` java
@Configuration
@EnableOAuth2Sso
public class MyConfig extends WebSecurityConfigurerAdapter {

    @Override
    protected void configure(HttpSecurity http) throws Exception {

        http.authorizeRequests()
                .antMatchers("/ui/**").authenticated()
                .anyRequest().permitAll()
                .and()
                .formLogin();

    }

}
```

My expectation would be that my configuration is enhanced with `OAuth2SsoCustomConfiguration`, which will add SSO specific behaviour to the `HttpSecurity` object. Instead, my tests showed that `OAuth2SsoDefaultConfiguration` is used, which additionally applies default request rules to the `HttpSecurity` object and hence messes up my intentional configuration.

The reason for this is, that the conditions in `OAuth2SsoCustomConfiguration` and `OAuth2SsoDefaultConfiguration`  are executed _before_ MyConfig is registered with the bean factory. The result is that the search for `EnableOAuth2Sso` annotated subclasses of `WebSecurityConfigurerAdapter` finds no matches and the implementation incorrectly assumes that I am _not_ using my own `WebSecurityConfigurerAdapter` .

**Comment by philwebb** [comment]
Could you please try again with the 1.5.0.RC1, if things are still failing could we please have a sample that reproduces the issue?

**Comment by jarnbjo** [comment]
Which changes have you done in 1.5.0.RC1, which are supposed to solve the problem? 

But no, I cannot try again with 1.5.0.RC1. After no reactions to this bug report for almost 5 months, we have long ago abandonded Spring Boot and now use other means of configuration. 

And if you need a sample yourself to verify a potential bugfix: What else do you need than the trivial, but complete configuration I quoted in the bug report, with which you can verify the erroneous behaviour?

**Comment by philwebb** [comment]
@jarnbjo Thanks anyway. Unfortunately we're only a small team so we can't always get to every issue in a timely manor. I'm glad that you found an alternative solution.

**Comment by jarnbjo** [comment]
Still, my question is valid: Which changes have you done in 1.5.0.RC1, which are supposed to solve the problem?

Have you fixed the issue, or are you just closing it because it is not a problem for me anymore? I mean, it is not a corner case I ran into, but as I wrote in the original issue description, the backing implementation of the @EnableOAuth2Sso  annotation is inherently broken and will not work together with configurations subclassed from WebSecurityConfigurerAdapter.
